src/modules/systemlib/mixer/mixer_multirotor.cpp

```diff
@@ -244,14 +244,23 @@ MultirotorMixer::mix(float *outputs, unsigned space)
 	if (min_out < 0.0f) {
 		float scale_in = thrust / (thrust - min_out);
 
+		max_out = 0.0f;
+
 		/* mix again with adjusted controls */
 		for (unsigned i = 0; i < _rotor_count; i++) {
-			outputs[i] = scale_in * (roll * _rotors[i].roll_scale + pitch * _rotors[i].pitch_scale) + thrust;
+			float out = scale_in * (roll * _rotors[i].roll_scale + pitch * _rotors[i].pitch_scale) + thrust;
+
+			/* update max output value */
+			if (out > max_out) {
+				max_out = out;
```

Comment:
What do you mean?
max_out value used later (https://github.com/PX4/Firmware/blob/95be7c22896571ca7555ee54eae7fac43cc867a4/src/modules/systemlib/mixer/mixer_multirotor.cpp#L271). So if we had low-end cropping (negative thrust on some rotors) we reduce roll/pitch controls, and must update max_out, so high-end limiting will work correctly.